File: 2013/UnSystem/Applications/DBAccess/LoginDBAccess.vb
Imports System.Configuration
Imports MySql.Data.MySqlClient

' ログイン用アクセスクラス
Public Class LoginDBAccess

    Private Connection As MySqlConnection
    Private Command As MySqlCommand
    Private DataReader As MySqlDataReader

    Public Sub New()
        ' DBの接続情報を生成
        Connection = New MySqlConnection
        Command = New MySqlCommand
        DataReader = Nothing

        If ConfigurationManager.AppSettings("ConnectString") IsNot Nothing Then
            ' Web.configよりコネクションを取得
            Connection.ConnectionString = ConfigurationManager.AppSettings("ConnectString").ToString
        End If
    End Sub

    Public Sub Dispose()
        ' クラス情報破棄
        Connection.Close()
        Command.Dispose()
        Connection.Dispose()
        If DataReader IsNot Nothing Then
            DataReader.Dispose()
        End If
    End Sub

    ''' <summary>
    ''' 認証チェック用データ取得
    ''' </summary>
    ''' <param name="strid">ユーザーID</param>
    ''' <param name="strpass">ユーザーパス</param>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Function GetTBLAuthCheck(ByVal strid As String, ByVal strpass As String) As MySqlDataReader

        ' オープン
        Connection.Open()

        Command = Connection.CreateCommand

        ' 認証用データ取得
        Command.CommandText = " SELECT * From un.trn_auth t1" & _
                              " INNER JOIN un.trn_personal t2 ON t1.PersonalID = t2.PersonalID " & _
                              " WHERE t1.PersonalID = " & strid & _
                              " AND " & " t1.PersonalPW = " & strpass

        ' コマンド送信
        DataReader = Command.ExecuteReader

        ' データ読み込み
        DataReader.Read()

        Return DataReader

    End Function

End Class

File: 2014/勤怠アプリ学習/WebApplication2/WebApplication2/DBAccess/LoginDBAccess.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Configuration;
using MySql.Data.MySqlClient;
using System.Data;

namespace WebApplication2.DBAccess
{
    public class LoginDBAccess
    {
        private MySqlConnection Connection;
        private MySqlCommand Command;

        //newコンストラクタ
        public LoginDBAccess()
        {
            if (ConfigurationManager.AppSettings["ConnectString"] != null)
            {
                //DB接続情報生成
                String conn = ConfigurationManager.AppSettings["ConnectString"].ToString () ;
                //Connection.ConnectionString = ConfigurationManager.AppSettings["ConnectString"];
                Connection = new MySqlConnection(conn);
                Command = new MySqlCommand();
            }
        }

        //ログインデータの取得
        public String[] getDataReader(string strID,string strPass)
        {
            Connection.Open();
            
            Command = Connection.CreateCommand();

            Command.CommandText = "SELECT PER.STAFF_ID,PER.FIRST_NAME,PER.LAST_NAME " +
                                  "FROM   unserver.personal_mst AS PER " +
                                  "INNER  JOIN unserver.parsonal_auth_trn AS PAS " +
                                  "ON     PER.STAFF_ID = PAS.STAFF_ID " +
                                  "WHERE  PAS.STAFF_ID = '" + strID +
                                  "' AND  PAS.PASSWORD = '" + strPass + "';";

            MySqlDataReader DataReader = Command.ExecuteReader();

            //データ読み込み
            //DataTable dt = new DataTable();
            //dt= DataReader.GetSchemaTable();

            DataReader.Read();
            string[] sessionString = new string[3];

            sessionString[0] = DataReader["STAFF_ID"].ToString ();
            sessionString[1] = DataReader["FIRST_NAME"].ToString();
            sessionString[2] = DataReader["LAST_NAME"].ToString();

            Connection.Close();

            //return DataReader;
            return sessionString;

        }
    }
}